skyllh/core/llhratio.py

```diff
@@ -1341,32 +1360,46 @@ def __call__(self, fitparam_values):
 
         Returns
         -------
-        f : (N_datasets,)-shaped 1D ndarray
-            The dataset signal weight factor for each dataset.
-        f_grads : (N_datasets,N_fitparams)-shaped 2D ndarray | None
-            The gradients of the dataset signal weight factors, one for each
-            fit parameter. None is returned if there are no fit parameters
-            beside ns.
+        f : (N_sources,)-shaped 1D ndarray
+            The source weight factor for each source.
+        f_grads : (N_sources,)-shaped 1D ndarray | None
+            The gradients of the source weight factors. None is returned if
+            there are no fit parameters beside ns.
         """
         fitparams_arr = self._src_fitparam_mapper.get_fitparams_array(fitparam_values[1:])
 
         N_fitparams = self._src_fitparam_mapper.n_global_fitparams
 
-        # Loop over the detector signal efficiency instances for the first and
-        # only source hypothesis group.
+        Y = []
+        Y_grads = []
 
-        # currently this only works for one source hypothesis group
-        Y, Y_grads = self.detsigyield_arr[0](self._src_arr_list[0], fitparams_arr)
+        # Loop over detector signal efficiency instances for each source
+        # hypothesis group in source hypothesis group manager.
+        for (j, detsigyield) in enumerate(self._detsigyield_arr):
```

Comment:
Change index to `g` (group) as `j` is used to index over datasets in the documentation.

Reply:
Fixed